### PR TITLE
Feature/material migration

### PR DESCRIPTION
**Pull request by aflegel**
Major overhaul to the material-ui react components.
Switched to a new graph provider
Added an advanced breakdown search.

### CODE REVIEW

File: src/Components/Layout/NavMenu.tsx [hunk 1]
```diff
@@ -1,21 +1,17 @@
 import React, { FunctionComponent } from "react";
-import { Link } from "react-router-dom";
+import AppBar from "@material-ui/core/AppBar";
+import Toolbar from "@material-ui/core/Toolbar";
+import Typography from "@material-ui/core/Typography";
```
[the remark this answers is not in the document]
Comment (aflegel) [reply src/Components/Layout/NavMenu.tsx]: I pulled these imports from the example pages.  This should be cleaned up.

[the remark this answers is not in the document]
Comment (aflegel) [reply src/Components/Layout/NavMenu.tsx]: I take it back, apparently the default export does not have proper constuctors.  I think it's a typing issue.

File: src/Components/Layout/Layout.tsx [hunk 2]
```diff
@@ -1,13 +1,11 @@
 import React, { FunctionComponent } from "react";
 import { NavMenu } from "./NavMenu";
 import { Footer } from "./Footer";
+import CssBaseline from "@material-ui/core/CssBaseline";
```
[the remark this answers is not in the document]
Comment (aflegel) [reply src/Components/Layout/Layout.tsx]: well, `CssBaseline` needs to be loaded in the component.

File: src/Components/Dice/Die.tsx [hunk 3]
```diff
@@ -9,27 +9,20 @@ export interface IDieProps {
...
-		case DieType.Setback:
-			dieSize = 6;
-			break;
-		case DieType.Challenge:
-		case DieType.Proficiency:
-		case DieType.Force:
-			dieSize = 12;
-			break;
-	}
+	const DieSize = () => {
```
[the remark this answers is not in the document]
Comment (aflegel) [reply src/Components/Dice/Die.tsx]: Habit.  I prefer functions and classes to be in pascal case.  If I could get properties to work nicely across the transpiler I'd probably pascal case those too.

[the remark this answers is not in the document]
Comment (ChrisDusyk) [reply src/Components/Dice/Die.tsx]: For sure, just commenting on it. Go with what you like!

**Pull request merged**